### PR TITLE
✨ Expose expectation value computation from DD Package to DDSIM

### PR DESCRIPTION
**Pull request by burgholzer**
This small PR exposes the newly added expectation value functionality from the DD Package to the `CircuitSimulator` in C++ as well as Python.
This allows to compute expectation values of observables that can be specified as quantum circuits. On the Python side, Qiskit `QuantumCircuit`s can be conveniently used.

### CODE REVIEW

File: src/CircuitSimulator.cpp [hunk 1]
```diff
@@ -83,6 +85,18 @@ std::map<std::string, std::size_t> CircuitSimulator<Config>::simulate(std::size_
     return measurementCounter;
 }
 
+template<class Config>
+dd::fp CircuitSimulator<Config>::expectationValue(const qc::QuantumComputation& observable) {
+    // simulate the circuit to get the state vector
+    simulate(0);
```
Comment (hillmich) [inline src/CircuitSimulator.cpp:91]: You might want to call `singleShot` directly, with `simulate(0)` you may end up not running the simulation altogether. (Or just call `simulate(1)`).

**Review by hillmich — CHANGES_REQUESTED** [review]
I have only one suggestion.

File: src/CircuitSimulator.cpp (hunk 1 above)
Comment (burgholzer) [reply src/CircuitSimulator.cpp]: Thanks again for the suggestion. Makes a lot of sense.
Done in https://github.com/cda-tum/ddsim/pull/203/commits/c084051726b89460037e599f19450a9899f0718d.

**Review by hillmich — APPROVED** [review]
LGTM!

**Pull request merged**